### ISSUE TITLE
Failed to verify a successful deployment if replicas set to other than 1

### ISSUE THREAD

**Issue by tjololo** [quay/quay-operator#158]
After deploying quay with redis replicas set to 3
```yaml
....
  redis:
    replicas: 3
....
```
We are seeing these logstatements in the controller:
```shell script
....
{"level":"info","ts":1583236234.4436715,"logger":"controller_quayecosystem","msg":"Reconciling QuayEcosystem","Request.Namespace":"quay-enterprise","Request.Name":"quay"}
{"level":"info","ts":1583236236.5694118,"logger":"controller_quayecosystem","msg":"Waiting for deployment","Namespace":"quay-enterprise","Name":"quay-redis","Timeout":420}
{"level":"error","ts":1583236656.5402331,"logger":"controller_quayecosystem","msg":"Failed to verify a successful deployment","Name":"quay-redis","stacktrace":"github.com/go-logr/zapr.(*zapLogger) .....
....
```
The controller moves on, but not before a timeout happens.
Same behavior observed if replicas != 1 for quay deployment 

The GetDeploymentStatus function assumes that the scale is always 1 for ok deployments.
```go
deployment.Status.AvailableReplicas == 1
```

I'm suspecting that this is the reason why it always fails to verify the success of the deployment

I'm more than happy to submit a PR with a proposal if you want

**Comment by tjololo** [comment]
Another question: should the check be change to verify the number of replicas against UpdatedReplicas or ReadyReplicas instead of AvailableReplicas as a rollout might be happening?

**Comment by tjololo** [comment]
#159 uses .Status.Replicas instead of 1. Not sure if this is the correct way.

**Comment by sabre1041** [comment]
Resolved as part of #187

**Comment by sabre1041** [comment]
@tjololo verification logic has been updated to address this issue
